src/queryCache.js

```diff
@@ -526,6 +581,34 @@ export function makeQueryCache() {
     return query
   }
 
+  if (initialQueries) {
+    for (const [queryHash, query] of Object.entries(initialQueries)) {
+      const queryKey = queryKeyParserFn(queryHash)
+      const queryConfig = query.config || {}
+
+      const config = {
+        ...defaultConfigRef.current,
+        ...queryConfig,
+      }
+
+      cache.queries[queryHash] = makeQuery({
+        cache,
+        queryKey,
+        queryHash,
+        config,
+      })
+
+      cache.queries[queryHash].wasPrefetched = true
+
+      if (!isServer) {
+        cache.queries[queryHash].scheduleStaleTimeout()
+        // Schedule for garbage collection in case
+        // nothing subscribes to this query
+        cache.queries[queryHash].scheduleGarbageCollection()
+      }
```

Comment:
Upon reflection, this block is a mistake. Side effects when creating things isn't very good practice and for Next for example we want `makeQueryCache` to be able to run inside of render.

One solution would be to move this into an init-function and let `ReactQueryCacheProvider` check if the cache has been/needs to be initialized, in that case that happens in an effect. A custom cache should always be paired with a cache provider, so this should provide an equally simple API for users, but be a safe way to do it.

Another option would be to add the prop `initialQueries` directly to the cache provider to avoid having to create a store at all on the client, which could be an ever nicer API?